baseplate/experiments/providers/variant_sets/multi_variant_set.py

```diff
@@ -0,0 +1,65 @@
+from .base import VariantSet
+
+
+class MultiVariantSet(VariantSet):
+    """ Variant Set designed to handle more than two total treatments.
+
+    MultiVariantSets are not designed to support changes in variant sizes without
+    rebucketing.
+    """
+
+    def __init__(self, variants, num_buckets=1000):
+        """ :param list variants: Array of dicts, each containing the keys 'name'
+            and 'size'. Name is the variant name, and size is the fraction of
+            users to bucket into the corresponding variant. Sizes are expressed
+            as a floating point value between 0 and 1.
+        :param int num_buckets: The number of potential buckets that can be
+            passed in for a variant call. Defaults to 1000, which means maximum
+            granularity of 0.1% for bucketing
+        """
+        self.num_buckets = num_buckets
+        self._validate_variants(variants)
+        self.variants = variants
+
+    def __contains__(self, item):
+        for variant in self.variants:
+            if variant.get('name') == item:
+                return True
+
+        return False
+
+    def _validate_variants(self, variants):
+
+        if variants is None:
+            raise ValueError('No variants provided')
+
+        if len(variants) < 3:
+            raise ValueError("MultiVariant experiments expect three or "
+                "more variants.")
+
+        total_size = 0.0
+        for variant in variants:
+            if variant.get('size') is None:
+                raise ValueError('Variant size not provided: {}'.format(variants))
+            total_size += int(variant.get('size') * self.num_buckets)
```

Comment:
I think it's better to make this a `@classmethod` and pass in `num_buckets` as an argument.  It's a little weird to have a function called in the constructor that expects some parts to be set up.  Another option that might be ok is to assign everything and then call `self._validate_variants` and use both `self.size` and `self.variants`.

Reply:
I like the latter approach. Calling validate first is a remnant of the older setup, where there was some processing on the variants themselves.